utensor_cgen/transformer/ns_transformer.py

```diff
@@ -13,7 +13,175 @@
+            tensor_size = self._getSize(out_o.shape)
+            if self._check(ugraph, allocate_table, use_def_table, out_o, i, i + tensor_size):
+              self._create_allocate_table(allocate_table, use_def_table, out_o, i, i + tensor_size)
+              success = True
+              break
+            else:
+              success = False
+        if success == False:
+          break
+      if success == True:
+        allocate_success = True
+    for node_name in ugraph.topo_order:
+      in_t_infos = ugraph.ops_info[node_name].input_tensors
+      for in_o in in_t_infos:
+        if in_o.name in allocate_table:
+          in_o.alloc_address[0] = allocate_table[in_o.name]['offsetstart']
+          in_o.alloc_address[1] = allocate_table[in_o.name]['offsetend']
+      out_t_infos = ugraph.ops_info[node_name].output_tensors
+      for out_o in out_t_infos:
+        if out_o.name in allocate_table:
+          out_o.alloc_address[0] = allocate_table[out_o.name]['offsetstart']
+          out_o.alloc_address[1] = allocate_table[out_o.name]['offsetend']
+    return ugraph
+
+
+  def _query_offset_fromallocate_table(self, allocate_table, start, end):
+    new_start = start
+    new_end = end
+    for key in allocate_table:
+      if allocate_table[key]['offsetstart'] >= start and allocate_table[key]['offsetend'] <= end:
+        continue
+      elif allocate_table[key]['offsetstart'] <= start and allocate_table[key]['offsetend'] >= start:
+        new_start = allocate_table[key]['offsetstart']
+        if allocate_table[key]['offsetend'] >= end:
+          new_end = max(new_end, allocate_table[key]['offsetend'])
+        else:
+          new_end = max(end, new_end)
+      elif allocate_table[key]['offsetstart'] >= start and allocate_table[key]['offsetend'] >= start:
+        if allocate_table[key]['offsetend'] >= end:
+          new_end = max(new_end, allocate_table[key]['offsetend'])
+        else:
+          new_end = max(end, new_end)
+    return new_start, new_end
+
+  def _query_time_fromallocate_table(self, allocate_table, start, end):
+    time_start = start
+    time_end = end
+    for key in allocate_table:
+      if allocate_table[key]['start'] >= start and allocate_table[key]['end'] <= end:
+        continue
+      elif allocate_table[key]['start'] <= start and allocate_table[key]['end'] >= start:
+        if allocate_table[key]['end'] >= end:
+          time_end = max(time_end, allocate_table[key]['end'])
+        else:
+          time_end = max(end, time_end)
+      elif allocate_table[key]['start'] >= start and allocate_table[key]['end'] >= start:
+        if allocate_table[key]['end'] >= end:
+          time_end = max(time_end, allocate_table[key]['end'])
+        else:
+          time_end = max(end, time_end)
+    return time_start, time_end
+
+  def _query_result(self, allocate_table, op, offset, length, timestart,
+  timeend):
+    ret = list()
+    for key in allocate_table:
+      '''print(key)
```

Comment:
Please remove these debugging comments in your PR.
It's annoying to read and in fact `pdb` should be a better option in python.
We are not writing C/C++.